### PR TITLE
add interfaces to CRL objects

### PR DESCRIPTION
**Pull request by etrauschke**
After a bunch of battles with git I think I have a considerably clean pull request.
I'm aware that this is not ready for putback yet since it's missing the updated doc part. I just wanted to ask if these docs are generated with a tool or if I just have to add my new interfaces to them manually.

The rest of the code should be ready, let me know if you have any concerns.

Erik

### CODE REVIEW

**Comment by alex** [comment]
Jenkins, add to whitelist

**Comment by alex** [comment]
Thanks for working on this! I think we're going to want to split into a few separate patches. The easiest to start with splitting out would be the bindings, and the interfaces themselves.

**Comment by etrauschke** [comment]
OK, I can do that. I'll create a new branch for this.

**Comment by etrauschke** [comment]
I'm already working on the coverage problem. The issue is that for some error cases it's hard to create certificates to test with. However, I now have a small C tool which can basically generate whatever CRL I like. I'll add this once I file the pull request for the backend fixes separately.

**Comment by AndreLouisCaron** [comment]
@etrauschke Thanks for working on this.  I have a functional implementation of a builder class for generating CRLs.  I have a minimal CRL interface to get that working, but I think I'll wait for this PR to be completed before I post the builder up for review.

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -1274,6 +1320,22 @@ def _public_key_bytes(self, encoding, format, evp_pkey, cdata):
         assert res == 1
         return self._read_mem_bio(bio)
 
+    def _parse_asn1_time(self, asn1_time):
```
Comment (AndreLouisCaron) [inline src/cryptography/hazmat/backends/openssl/backend.py]: I recently moved similar methods to free functions and renamed them to `_decode_*`.  Maybe you can align with that?

**Comment by AndreLouisCaron** [comment]
> I just wanted to ask if these docs are generated with a tool or if I just have to add my new interfaces to them manually.

You'll have to add the interfaces to the documentation manually.  The file to edit is `docs/x509.rst`.  You're probably going to want to look at [Sphinx](http://sphinx-doc.org/) to know how to write them, but you can also take a look at two of my recent PRs which add interfaces related to the ones in this PR:
- #1927 CSR builder
- #1980 Certificate builder

File: src/cryptography/hazmat/backends/openssl/backend.py (hunk 1 above)
Comment (etrauschke) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Ok, I can wait for this to go back first. Then I'll rebase my changes for final review.

**Comment by reaperhulk** [comment]
Does #1999 replace this or should I keep this open?

**Comment by etrauschke** [comment]
Yes. Basically I split this PR into multiple sub-PRs. This PR can be closed.

Erik

**Comment by alex** [comment]
Cheers!

**Pull request closed without merging**